packages/server/lib/unhandled_exceptions.ts

```diff
@@ -0,0 +1,26 @@
+import Debug from 'debug'
+const debug = Debug('cypress:server:unhandled_exceptions')
+
+export function handle (shouldExitCb?: (err: Error) => boolean) {
+  function globalExceptionHandler (err: Error) {
+    if (shouldExitCb && !shouldExitCb(err)) {
+      debug('suppressing unhandled exception, not exiting %o', { err })
+      handle(shouldExitCb)
+
+      return
+    }
+
+    process.exitCode = 1
+
+    return require('./errors').logException(err)
```

Comment:
I wanted to deviate from the existing implementation as little as possible. We've had regressions due to incorrect exitCode before so I added this line, but otherwise didn't modify the default unhandled exception behavior.

But since you ask... this is a weird one. If a handler is registered for unhandled exceptions, NodeJS won't automatically run `process.exit`. Normally this would happen synchronously, and `.on('exit')` is actually *supposed* to be synchronous as well - the handler is not `await`ed. We are sort of assuming that nothing will cause a `process.exit(0)` between the asynchronous `logException` call and `process.exit` here.

Reply:
let's leave it then 👍